traffic-sniffer/lib/sniffer/mock/mock-manager.ts

```diff
@@ -25,12 +25,31 @@ export default class MockManager {
 
     if (managedMock !== undefined) {
       managedMock.data = data;
+
       this.mocks.set(id, managedMock);
     } else {
       throw new MockNotFoundError();
     }
   }
 
+  updateMock(id: string, mock: Mock) {
+    let managedMock = this.mocks.get(id);
+
+    if (managedMock !== undefined) {
+      const newId = `${mock.method} ${mock.endpoint}`;
+      managedMock = {
+        ...managedMock,
+        ...mock,
+        id: newId,
+      };
+
+      this.mocks.delete(id);
```

Comment:
Is it necessary to delete?

Reply:
The old mock and the new mock share the same Id

Reply:
why don't we use uuid or nanoid to generate a unique id

Reply:
We have uuid in some places in the system. But here its not relevant. because what identifies a mock is the endpoint. 

METHOD /path/of/mock